int32[]  robot_pos_x                                      
int32[]  robot_pos_y                               
int32[]  task_pos_x
int32[]  task_pos_y
bool     is_noise                                  
